/work3/soeba/HALOS/halos/lib/python3.8/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and will be removed in 0.15, please use 'weights' instead.
  warnings.warn(
/work3/soeba/HALOS/halos/lib/python3.8/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and will be removed in 0.15. The current behavior is equivalent to passing `weights=VGG19_Weights.IMAGENET1K_V1`. You can also use `weights=VGG19_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)
Training Epoch 0
Trained batch 2369 in epoch 0, gen_loss = 2.4674548710197337, disc_loss = 0.016571816910851064
Testing Epoch 0

Training Epoch 1
Trained batch 2369 in epoch 1, gen_loss = 2.26309132910479, disc_loss = 0.04179788297744672
Testing Epoch 1
Training Epoch 2
Trained batch 2369 in epoch 2, gen_loss = 2.011633070171634, disc_loss = 0.0835847217645614
Testing Epoch 2
Training Epoch 3
Trained batch 2369 in epoch 3, gen_loss = 2.026856710601457, disc_loss = 0.0767194268002886
Testing Epoch 3
Training Epoch 4
Trained batch 2369 in epoch 4, gen_loss = 2.083367960538542, disc_loss = 0.06633478021779675
Testing Epoch 4

  0%|          | 0/25 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/work3/soeba/HALOS/halos/lib/python3.8/site-packages/PIL/Image.py", line 2928, in fromarray
    mode, rawmode = _fromarray_typemap[typekey]
KeyError: ((1, 1, 128), '|u1')
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "esrgan_bones.py", line 417, in <module>
    im = Image.fromarray(np.uint8(gen_hr[i].cpu().detach()*255)).convert("L")
  File "/work3/soeba/HALOS/halos/lib/python3.8/site-packages/PIL/Image.py", line 2930, in fromarray
    raise TypeError("Cannot handle this data type: %s, %s" % typekey) from e
TypeError: Cannot handle this data type: (1, 1, 128), |u1